README.md

```diff
@@ -349,6 +318,39 @@ const result = err(12)
 const twelve = result._unsafeUnwrapErr()
 ```
 
+
+---
+
+### `Result.match` (method)
+
+Given 2 functions (one for the `Ok` variant and one for the `Err` variant) execute the function that matches the `Result` variant.
+
+Both functions will unwrap the `Result` object and must have the same return type.
+
+`match` is sort of like combining `_unsafeUnwrap` and `_unsafeUnwrapErr`.
```

Comment:
Thanks for the PR! 

> `match` is sort of like combining `_unsafeUnwrap` and `_unsafeUnwrapErr`.

Regarding this comment, note that the analogy is more accurate when you use `map` and `mapErr`. 

The reason is that `match` is a alternative api to chaining `map` and `mapErr`. 

```typescript
import { ok, Result } from 'neverthrow'

const myval: Result<number, string> = ok(100)

const successCallback = (someNumber: number) => {
  console.log('> number is: ', someNumber)
}

const failureCallback = (someFailureValue: string) => {
  console.log('> boooooo')
}

// method chaining api
// note that you DONT have to append mapErr
// after map which means that you are not required to do
// error handling
myval
  .map(successCallback)
  .mapErr(failureCallback)


// match api
// works exactly the same as above,
// except, now you HAVE to do error handling :)
myval.match(
  successCallback,
  failureCallback
)
```

You can play with the source here:

https://repl.it/repls/FrugalMajorRedundantcode

One tiny distinction between these two similar approaches is that with method chaining, your callbacks don't need to have the same return type, whereas with match you do!

So my ask is that you change that sentence to maybe include the above explanation, or something along those lines. I realize in retrospect that I should have been more clear when I wrote, "`match` is sort of like combining `map` and `mapErr`".

Reply:
Thanks for the feedback, and for clarifying the match method.
I'll apply this changes right away !